fclient/federationclient_test.go

```diff
@@ -58,7 +58,7 @@ func TestSendJoinFallback(t *testing.T) {
 		},
 		fclient.WithSkipVerify(true),
 	)
-	fc.Client = *fclient.NewClient(fclient.WithTransport(
+	fc.SetInternalClient(*fclient.NewClient(fclient.WithTransport(
```

Comment:
We should be able to use `fc.NewFederationClient(signingIdentity, fclient.WithTransport(...))` instead of gut wrenching the client?